metacat-connector-druid/src/main/java/com.netflix.metacat.connector.druid/DruidConnectorFactory.java

```diff
@@ -0,0 +1,116 @@
+import com.netflix.metacat.common.server.connectors.ConnectorFactory;
+import com.netflix.metacat.common.server.connectors.ConnectorPartitionService;
+import com.netflix.metacat.common.server.connectors.ConnectorTableService;
+import com.netflix.metacat.connector.druid.configs.DruidConnectorConfig;
+import com.netflix.metacat.connector.druid.configs.DruidHttpClientConfig;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.core.env.MapPropertySource;
+import org.springframework.core.env.MutablePropertySources;
+import org.springframework.core.env.StandardEnvironment;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Druid Connector Factory.
+ *
+ * @author zhenl
+ * @since 1.1.0
+ */
+public class DruidConnectorFactory implements ConnectorFactory {
+    private final String catalogName;
+    private AnnotationConfigApplicationContext ctx;
+
+    /**
+     * Constructor.
+     *
+     * @param catalogName      connector name. Also the catalog name.
+     * @param connectorContext connector config
+     */
+    DruidConnectorFactory(
+        final String catalogName,
+        final ConnectorContext connectorContext
+    ) {
+        this.catalogName = catalogName;
+        this.ctx = new AnnotationConfigApplicationContext();
+        this.ctx.getBeanFactory().registerSingleton("ConnectorContext", connectorContext);
+        final StandardEnvironment standardEnvironment = new StandardEnvironment();
+        final MutablePropertySources propertySources = standardEnvironment.getPropertySources();
+        final Map<String, Object> properties = new HashMap<>();
+        if (Boolean.parseBoolean(connectorContext.getConfiguration().get("httpClient"))) {
+            properties.put("httpClient", true);
+        }
+        propertySources.addFirst(new MapPropertySource("DRUID_CONNECTOR", properties));
+        this.ctx.setEnvironment(standardEnvironment);
+        this.ctx.register(DruidConnectorConfig.class);
+        this.ctx.register(DruidHttpClientConfig.class);
+        this.ctx.refresh();
+    }
+
+    /**
+     * Returns the database service implementation of the connector.
+     *
+     * @return Returns the database service implementation of the connector.
+     */
+    @Override
+    public ConnectorDatabaseService getDatabaseService() {
+        return this.ctx.getBean(DruidConnectorDatabaseService.class);
+    }
+
+    /**
+     * Returns the table service implementation of the connector.
+     *
+     * @return Returns the table service implementation of the connector.
+     */
+    @Override
+    public ConnectorTableService getTableService() {
+        return this.ctx.getBean(DruidConnectorTableService.class);
+    }
+
+    /**
+     * Returns the partition service implementation of the connector.
+     *
+     * @return Returns the partition service implementation of the connector.
+     */
+    @Override
+    public ConnectorPartitionService getPartitionService() {
```

Comment:
Is there an equivalent of 'partition' in druid? If there is none, then throw NotSupportedException

Reply:
This is a good point. There's no partition, table, or database concept in druid.  For now, the data source is treated as a table, but no partition is introduced. The segment can be treated as partitions conceptually. Would like to understand the application query, if they are only interested in the latest one, there's no reason to introduce the partition concepts. But they want to query individual segments, we will use partitions to manage them.